pkg/drivers/bess/reader.go

```diff
@@ -0,0 +1,144 @@
+//Copyright (c) 2019 Red Hat Inc
+//
+//License: GPL2, see COPYING in source directory
+
+
+package bess
+
+
+import (
+    "net"
+	"github.com/nimbess/nimbess-agent/pkg/network"
+    log "github.com/sirupsen/logrus"
+    "time"
+    "github.com/google/gopacket"
+    "github.com/google/gopacket/layers"
+)
+
+
+const cutoff = 300
+
+type Reader struct {
+    c  net.Conn
+    buffer  []byte
+    port string
+    path  string
+    macs map[string]int64
+    control chan network.L2FIBCommand
+}
+
+func NewReader (portArg string, pathArg string, controlChan chan network.L2FIBCommand) (*Reader) {
+    r := &Reader{
+        port: portArg,
+        path: pathArg,
+        macs: make(map[string]int64),
+        buffer: make([]byte, 2048),
+        control: controlChan,
+    }
+    for i := 0; i < 10; i++ {
+        c, err := net.Dial("unixpacket", r.path)
+        if (err == nil) {
+            r.c = c
+            return r
+        }
+        if err != nil {
+            log.Debugf("Could Not connect to %s, error %s, retry no %d in one second:", r.path, err, i)
+        }
+        time.Sleep(time.Second)
+    }
+    log.Fatal("Could Not connect:")
```

Comment:
Please change this to log.Error and return a nil reader or error to the calling function and handle it there. This is important so we handle the CNI reply gracefully that it failed and not just exit the program

Reply:
also need to add the unix socket path (shared-dir) to the mount for the agent in nimbess.yaml